### DISCUSSION TITLE
getting this error 'box2d/box2d.h' file not found with minSdkVersion 21

### DISCUSSION THREAD

**Discussion by meehjeevan** [axmolengine/axmol#1958] [Q&A] [answered]
So, I need to set the minSdkVersion to 21 because the latest google ad sdk requires it to be 21. 
(com.google.android.gms:play-services-ads:23.1.0)

While compiling, I got this error:

```
/axmol/extensions/physics-nodes/src/physics-nodes/PhysicsDebugNodeBox2D.h:26:10: fatal error: 'box2d/box2d.h' file not found
#include "box2d/box2d.h"
         ^~~~~~~~~~~~~~~
```
But when I compile with less than 21 value and without play services-ads it works without the errors.

And interesting thing is this error is occuring on iOS version from the beginning in mine.

**Comment by rh101** [comment]
@meehjeevan You didn't need to create a new thread for this, since you already had another 2:
https://github.com/axmolengine/axmol/discussions/1935
and 
https://github.com/axmolengine/axmol/discussions/1951

**Reply by meehjeevan** [reply]
The case is a bit different, so I created new post. In earlier, it was about ios build but I found it in android build too.

**Reply by rh101** [reply]
~~It's all about issues you are having with Xcode and iOS builds, so it would have been best to keep them in a single thread.~~

**Reply by meehjeevan** [reply]
No, sir, this one is on android with minsdkversion 21.

**Reply by rh101** [reply]
Oh wow, I'm so sorry, I completely misread. Never mind!

**Comment by rh101** [comment]
Is the project you're building a clean new project, or one you have ported across to Axmol?

If it's not a new project, can you please try creating a new project and building it to see if it has the same issues? It's the same question I asked in the other thread regarding iOS, but you never replied to it: https://github.com/axmolengine/axmol/discussions/1935#discussioncomment-9638717

**Reply by smilediver** [reply]
@meehjeevan also please provide the full command line that is being used to compile the failing cpp file.

**Reply by meehjeevan** [reply]
No command line..
I'm using Android Studio latest version and Xcode 12.4

**Reply by meehjeevan** [reply]
> Is the project you're building a clean new project, or one you have ported across to Axmol?
> 
> If it's not a new project, can you please try creating a new project and building it to see if it has the same issues? It's the same question I asked in the other thread regarding iOS, but you never replied to it: https://github.com/axmolengine/axmol/discussions/1935#discussioncomment-9638717

I created new project in axmol and ported coco2dx 3.17.2 project sources into the newly created axmol project with manual changes.

**Reply by rh101** [reply]
@meehjeevan Can you please take a moment to simply generate a new project, but do not add any code to it.  Build that project to check if it works in both Xcode 12.4 and Android Studio, and then please reply back in this thread to whether it works or not.

**Reply by smilediver** [reply]
> No command line.. I'm using Android Studio latest version and Xcode 12.4

Android Studio behind the scenes will use some commands to compile cpp files, you need to find the command line used to compile the file that fails.

In Android Studio project folder recursively search for `compile_commands.json` file. If the file is not there, then add `set(CMAKE_EXPORT_COMPILE_COMMANDS ON)` to your `CMakeLists.txt` file, resync and recompile the project and then look for the `compile_commands.json` file again.

Inside the `compile_commands.json` you will find command lines used to compile each cpp file. Find the entry for the failing cpp file.

**Reply by meehjeevan** [reply, marked as answer]
> @meehjeevan Can you please take a moment to simply generate a new project, but do not add any code to it. Build that project to check if it works in both Xcode 12.4 and Android Studio, and then please reply back in this thread to whether it works or not.

The freshly created new blank project works well on android. I figured it out in my codes I have used this:
#include "extensions/cocos-ext.h" in many files, with this when minSdkVersion is 21 it was giving that errors. But I removed it and now the errors are gone and my game is also running.
